server/src/main/java/com/hedera/block/server/config/BlockNodeContextFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.config;
+
+import com.hedera.block.server.metrics.MetricsService;
+import com.swirlds.common.metrics.platform.DefaultMetricsProvider;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import com.swirlds.config.extensions.sources.ClasspathFileConfigSource;
+import com.swirlds.config.extensions.sources.SystemEnvironmentConfigSource;
+import com.swirlds.config.extensions.sources.SystemPropertiesConfigSource;
+import com.swirlds.metrics.api.Metrics;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class BlockNodeContextFactory {
+    private static final System.Logger logger =
+            System.getLogger(BlockNodeContextFactory.class.getName());
+
+    private static final String APPLICATION_PROPERTIES_1 = "app.properties";
+
+    private BlockNodeContextFactory() {}
+
+    public static BlockNodeContext create() {
+        final Configuration configuration = getConfiguration();
+        final Metrics metrics = getMetrics(configuration);
+        final MetricsService metricsService = new MetricsService(metrics);
+        return new BlockNodeContext(metrics, metricsService, configuration);
+    }
+
+    private static Configuration getConfiguration() {
+        try {
+            return ConfigurationBuilder.create()
+                    .withSource(SystemEnvironmentConfigSource.getInstance())
+                    .withSource(SystemPropertiesConfigSource.getInstance())
+                    .withSource(new ClasspathFileConfigSource(Path.of(APPLICATION_PROPERTIES_1)))
+                    .autoDiscoverExtensions()
+                    .build();
+        } catch (IOException e) {
+            logger.log(System.Logger.Level.ERROR, "Error reading configuration", e);
```

Comment:
The code that calls this method from `Server` is wrapped in a `try/catch` which catches `IOException`.  Maybe rethrow the IOException here rather than a RuntimeException so we have a consistent exit point?

Reply:
It looks like `ClasspathFileConfigSource` can throw an IOException from the constructor. Maybe we should pass this as an instance of ConfigSource to the `create()` method. In this way, we can mock it from the tests and ensure we get the correct behavior and logging if `ClasspathFileConfigSource` cannot find the `APPLICATION_PROPERTIES_1` file.

Reply:
✅  Done!